tools/walletextension/storage/storage.go

```diff
@@ -0,0 +1,69 @@
+package storage
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/obscuronet/go-obscuro/go/rpc"
+)
+
+type Storage struct {
+	db *SqliteDatabase
+}
+
+func New(dbPath string) *Storage {
+	// If path is empty we create a random throwaway temp file, otherwise we use the path to the database
+	if dbPath == "" {
+		tempDir := filepath.Join("/tmp", "obscuro_gateway", randomStr(8))
+		err := os.MkdirAll(tempDir, os.ModePerm)
+		if err != nil {
+			fmt.Println("Error creating directory: ", tempDir, err)
+		}
+		dbPath = filepath.Join(tempDir, "gateway_databse.db")
+	} else {
+		dir := filepath.Dir(dbPath)
+		err := os.MkdirAll(dir, 0o755)
+		if err != nil {
+			fmt.Println("Error creating directories:", err)
+		}
+	}
+
+	newDB, err := NewSqliteDatabase(dbPath)
+	if err != nil {
+		fmt.Println("Error creating database:", err)
+	}
+
+	return &Storage{db: newDB}
+}
+
+func (s *Storage) SaveUserVK(userID string, vk *rpc.ViewingKey) error {
+	err := s.db.Save(userID, vk)
+	if err != nil {
+		return fmt.Errorf("failed to save viewingkey to the storage, %w", err)
+	}
+	return nil
+}
+
+func (s *Storage) GetUserVKs(userID string) (map[common.Address]*rpc.ViewingKey, error) {
+	userVKs, err := s.db.Get(userID)
+	if err != nil {
+		return nil, err
+	}
+	return userVKs, nil
+}
+
+// Generates a random string n characters long.
+func randomStr(n int) string {
```

Comment:
could this go in a util file from the common package?
This is the second time we implement it

Reply:
Moved to common package